lib/JsonLdDocumentLoader.js

```diff
@@ -15,26 +15,52 @@ export class JsonLdDocumentLoader {
    *
    * @param {string} url - The URL of a document,
    *   etc.
-   * @param {object} document -The document.
+   * @param {object} document - The document.
+   * @param {object} [options] - The options to use.
+   * @param {boolean} [options.overwrite=false] - True to allow overwrites of
+   *   existing mappings.
    */
-  addStatic(url, document) {
+  addStatic(url, document, options) {
+    options = options || {};
     if(!_isString(url)) {
       throw new TypeError('The first parameter (url) must be a string.');
     }
     if(!_isObject(document)) {
       throw new TypeError('The second parameter (document) must be an object.');
     }
+    if(!options.overwrite) {
+      if(this.documents.has(url)) {
+        throw new Error(`Document mapping already exists: "${url}".`);
+      }
+    }
     this.documents.set(url, structuredClone(document));
   }
 
+  /**
+   * Adds many URL to document mappings to the loader.
+   *
+   * The documents will be cloned.
+   *
+   * @param {object} documents - An object that implements the iterable
+   *   protocol returning [url, document] pairs.
+   * @param {object} [options] - The options to use.
+   * @param {boolean} [options.overwrite=false] - True to allow overwrites of
+   *   existing mappings.
+   */
+  addDocuments(documents, options) {
```

Comment:
While the previous API was too old to use named parameters, I think this new API should use them.

```suggestion
  addDocuments({documents, options} = {}) {
```